### ISSUE TITLE
Predict all classes

### ISSUE THREAD

**Issue by sevinjyolchuyeva** [patil-suraj/exploring-T5#4]
I have applied your code on the emotion recognition task.  I have got a really good result. Thank you for this source.  I have 8 emotions and for test data, I would like to print each emotion with its probability, like ['happy': 0.061, 'surprise': 0.148, 'fear': 0.031]. In the prediction, only printed the only 1 emotion. I have increased the parameter of "num_return_sequences" to 8, then I have got some outputs which are not from emotions list. I know my question is a bit stranger :)

**Comment by patil-suraj** [comment]
Hi @sevinjyolchuyeva , num_return_sequences is not the solution here. It's meant for generation but this problem is not purely generative. To get the probability of the class tokens, make sure each class label is encoded as single id. Then to get get probability you'll need to generate yourself without using the `.generate` method and get the final logits. Then let's say we are predicting "true" or "false", to get probability apply softmax only on the "true" and "false" tokens (in your case, 8 tokens).

Hope this helps. I'm not entirely sure how this will work, but worth giving it a try

**Comment by sevinjyolchuyeva** [comment]
Hi, @patil-suraj :), 
Thank you so much for your comment. I have understood what you said. I will try it.
